app/javascript/shared/components/focusTrap.jsx

```diff
@@ -0,0 +1,30 @@
+import PropTypes from 'prop-types';
+import { h, Fragment } from 'preact';
+import { useLayoutEffect, useRef } from 'preact/hooks';
+import { createFocusTrap } from 'focus-trap';
+import { defaultChildrenPropTypes } from '../../common-prop-types';
+
+const FocusTrap = ({ selector, children }) => {
+  const focusTrap = useRef(null);
+
+  useLayoutEffect(() => {
+    focusTrap.current = createFocusTrap(selector);
+    focusTrap.current.activate();
+    return () => {
+      focusTrap.current.deactivate();
+    };
+  });
+
+  return <Fragment>{children}</Fragment>;
+};
+
+FocusTrap.defaultProps = {
+  selector: '.crayons-modal',
```

Comment:
nice, I like that you've made the most likely use-case the default :)

Reply:
refactor (non-blocking): The focus trap is working by wrapping `<main />` right now, but I can see it being even more useful in the context of the modal, i.e. embedded in the `<Modal />` component. This way modals, would always have the focus trap by default. This is definitely for another PR though. Also, if we were to do this, I would suggest dynamically importing a modal wherever we use them since they're only needed when requested and someone might never request a modal.